packages/shared/lib/sdk/sync.ts

```diff
@@ -59,6 +63,33 @@ interface DataResponse {
     [index: string]: unknown | undefined | string | number | boolean | Record<string, string | boolean | number | unknown>;
 }
 
+export enum PaginationType {
+    CURSOR = 'cursor',
+    LINK = 'link',
+    OFFSET = 'offset'
+}
+
+interface Pagination {
+    type: string;
+    limit?: number;
+    response_data_path?: string;
+    limit_parameter_name: string;
+}
+
+export interface CursorPagination extends Pagination {
+    next_cursor_parameter_path: string;
+    cursor_parameter_name: string;
+}
+
+export interface NextUrlPagination extends Pagination {
```

Comment:
Both of these params define the path in response.
If `link_rel` is passed, we will extract the next URL from `link` header and then we will extract value with specific relationship(in most cases it's `next`). See [GH pagination docs](https://docs.github.com/en/rest/guides/using-pagination-in-the-rest-api?apiVersion=2022-11-28) for example.
if `link_body_parameter_path`  is passed, we will extract next url from body. This is similar to cursor.

As I mentioned above, I do like `in_response` postfix more than `path` one so I'd update these like this:
- `link_path_in_response_headers`
- `link_path_in_response`

Reply:
One thing I'd note is that `link_path_in_response_headers` is not like all the other `path` fields out there since it simply represents the relationship of the URL that we should get from `link` header. I guess it might also make sense to have it called `link_rel_in_response_header` since this name reveals the intent better.

Reply:
Agreed, with all of that. I think it'd be great to have a quick sync on naming before shipping to harmonize everything.

Reply:
> Agreed, with all of that. I think it'd be great to have a quick sync on naming before shipping to harmonize everything.

Yep, I'll post the latest version summary once this PR has been approved or once it's close to that point so we all can take a final look.